samples/ImplicitFlow/ResourceServer02/Startup.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using NWebsec.AspNetCore.Middleware;
+
+namespace Mvc.Server
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseDeveloperExceptionPage();
+
+            app.UseStaticFiles();
+
+            // Add a middleware used to validate access tokens and protect the API endpoints.
+            // app.UseOAuthValidation();
+
+            // Alternatively, you can also use the introspection middleware.
+            // Using it is recommended if your resource server is in a
+            // different application/separated from the authorization server.
+
+            app.UseOAuthIntrospection(options =>
+            {
+                options.AutomaticAuthenticate = true;
+                options.AutomaticChallenge = true;
+                options.Authority = "http://localhost:12345/";
+                options.Audiences.Add("ResourceServer02");
+                options.ClientId = "ResourceServer02";
+                options.ClientSecret = "secret_secret_secret";
+            });
+
+            app.UseCsp(options => options.DefaultSources(directive => directive.Self())
```

Comment:
Content Security Policy: http://www.html5rocks.com/en/tutorials/security/content-security-policy/

You can remove it, it's not necessary for a resource server.